app-backend/common/src/main/scala/package.scala

```diff
@@ -0,0 +1,52 @@
+package com.azavea.rf
+
+import cats.data.NonEmptyList
+import com.azavea.rf.tool.eval.DatabaseError
+import java.util.UUID
+
+import scala.concurrent.{ExecutionContext, Future}
+
+import cats._
+import cats.data.OptionT
+import cats.data.Validated.{Invalid, Valid}
+import cats.implicits._
+import com.azavea.rf.database.Database
+import com.azavea.rf.database.tables.{ToolRuns, Tools}
+import com.azavea.rf.datamodel.User
+import com.azavea.rf.tool.ast.MapAlgebraAST
+import com.azavea.rf.tool.eval.Interpreter
+import com.azavea.rf.tool.params.EvalParams
+
+package object common {
+
+  /** Convert an [[Either]] to an [[Option]], or throw the error. */
+  def maybeThrow[A <: Throwable, B](e: Either[A, B]): Option[B] = e match {
+    case Right(a) => Some(a)
+    case Left(failure) => throw failure
+  }
+
+  /** Validate an AST, given some ToolRun. In the case of success, returns
+    * the zero element of some specified Monoid.
+    */
+  def validateAST[M: Monoid](
```

Comment:
The even smarter thing to do here might be to do the IO in the route, and pass the `toolRun` and `tool` into this function, then have it be extra pure. Then we could use straight `Either`, and avoid `Future` altogether.